### DISCUSSION TITLE
Share a simple code snippet

### DISCUSSION THREAD

**Discussion by wezmag** [serenity-is/Serenity#6662] [Tricks and Samples]
Hi,

I would like to share a simple code snippet that I've been using since I started using Serenity.
![fprop](https://user-images.githubusercontent.com/340794/213339858-6cffd697-436e-486a-b223-48d807782743.gif)

## Add Code Snippet to your Visual Studio
1. Create a code sinppet file with filename `FieldProp.snippet` and content as below:
``` xml
<?xml version="1.0" encoding="utf-8"?>
<CodeSnippets xmlns="http://schemas.microsoft.com/VisualStudio/2005/CodeSnippet">
	<CodeSnippet Format="1.0.0">
		<Header>
			<Title>Serenity Field Property</Title>
			<Author>Wesley Huang</Author>
			<Shortcut>fprop</Shortcut>
		</Header>
		<Snippet>
			<Code Language="CSharp">
            <![CDATA[
                [DisplayName("$DisplayName$")]
                public $Type$ $FieldName$
                {
                    get => fields.$FieldName$[this];
                    set => fields.$FieldName$[this] = value;
                }
            ]]></Code>
			<Declarations>
				<Literal>
					<ID>FieldName</ID>
					<ToolTip>Field Name</ToolTip>
					<Default>FieldName</Default>
				</Literal>
				<Literal>
					<ID>DisplayName</ID>
					<ToolTip>Display Name</ToolTip>
					<Default>DisplayName</Default>
				</Literal>
				<Literal>
					<ID>Type</ID>
					<ToolTip>Type</ToolTip>
					<Default>string</Default>
				</Literal>
			</Declarations>
		</Snippet>
	</CodeSnippet>
</CodeSnippets>
```
2. Place the file in the `\Documents\Visual Studio 2022\Code Snippets\Visual C#\My Code Snippets` folder.

## To use the code snippet.
1. type `fprop` and press `Tab` twice.
2. using `Tab` to switch between DisplayName, Type and FieldName
3. press `Enter` to add it to your code.

Hope you find this helpful.
Happy Lunar New Year!

**Comment by minhhungit** [comment]
Happy Lunar New Year, thank you for sharing
